### PR TITLE
CME-22 update to Spring Boot 3

### PR DESCRIPTION
**Pull request by rapolaskaseliscgi**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/CME-22

### Change description ###
Update to Spring Boot 3

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/reform/notifications/config/security/SpringSecurityConfiguration.java [hunk 1]
```diff
@@ -17,77 +16,75 @@
 import org.springframework.security.oauth2.jwt.JwtTimestampValidator;
 import org.springframework.security.oauth2.jwt.NimbusJwtDecoder;
 import org.springframework.security.oauth2.server.resource.authentication.JwtAuthenticationConverter;
-import org.springframework.security.oauth2.server.resource.web.BearerTokenAuthenticationFilter;
+import org.springframework.security.oauth2.server.resource.web.authentication.BearerTokenAuthenticationFilter;
+import org.springframework.security.web.SecurityFilterChain;
 import uk.gov.hmcts.reform.authorisation.filters.ServiceAuthFilter;
-
```
Comment (kiran-yenigala-hmcts) [inline src/main/java/uk/gov/hmcts/reform/notifications/config/security/SpringSecurityConfiguration.java:22]: https://learn.microsoft.com/en-us/azure/azure-monitor/app/java-spring-boot

Please look into the Application Insights with Spring boot 3

**Review by kiran-yenigala-hmcts — COMMENTED** [review]
Couple of comments added

File: build.gradle [hunk 2]
```diff
@@ -136,7 +124,7 @@ jacocoTestReport {
   reports {
     xml.required = true
     csv.required = false
-    xml.destination file("${buildDir}/reports/jacoco/test/jacocoTestReport.xml")
+    xml.outputLocation = file("${buildDir}/reports/jacoco/test/jacocoTestReport.xml")
```
[the remark this answers is not in the document]
Comment (rapolaskaseliscgi) [reply build.gradle:127]: No, it is not used anywhere

File: src/main/java/uk/gov/hmcts/reform/notifications/config/security/SpringSecurityConfiguration.java (hunk 1 above)
Comment (rapolaskaseliscgi) [reply src/main/java/uk/gov/hmcts/reform/notifications/config/security/SpringSecurityConfiguration.java:22]: Thanks, had a look.

File: src/main/java/uk/gov/hmcts/reform/notifications/config/AppInsightsConfiguration.java [hunk 3]
```diff
@@ -0,0 +1,13 @@
+package uk.gov.hmcts.reform.notifications.config;
+
+import com.microsoft.applicationinsights.TelemetryClient;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+
+@Configuration
+public class AppInsightsConfiguration {
```
Comment (kiran-yenigala-hmcts) [inline src/main/java/uk/gov/hmcts/reform/notifications/config/AppInsightsConfiguration.java:8]: why is this needed now?

Comment (rapolaskaseliscgi) [reply src/main/java/uk/gov/hmcts/reform/notifications/config/AppInsightsConfiguration.java]: Good spot, it is not needed. Removed it.

**Review by kiran-yenigala-hmcts — APPROVED** [review]
LGTM

**Review by Thor-tech-of-metal — APPROVED** [review]

**Review by davejones74 — APPROVED** [review]
